### PR TITLE
Edits to security authentication mechanisms doc

### PR DESCRIPTION
**Pull request by michelle-purcell**
To enhance and apply style as per the [Quarkus docs contributor style guide](https://quarkus.io/guides/doc-reference), mainly:
- One sentence per line.
- Converting `<<>>` to `xrefs:` to ensure they work downstream  (asciidoctorj & other PV2 transform dependencies)
- Fix table to work downstream  (asciidoctorj & other PV2 transform dependencies)
- Consistent capitalization in references to 'Bearer authentication' and the more generic `bearer token` term.

@sberyozkin - et al security folks - Do we need to add/update any of this content for the impending 3.2 release? You can add to this PR or tag me in your PRs and I'll help with any editing work that's needed. Thanks :smile:

### CODE REVIEW

**Comment by sberyozkin** [comment]
Thanks @michelle-purcell I'll have a look tomorrow

File: docs/src/main/asciidoc/security-authentication-mechanisms-concept.adoc [hunk 1]
```diff
@@ -150,64 +147,67 @@ X509Certificate certificate = credential.getCertificate();
...
-The Bearer Token mechanism extracts the token from the HTTP Authorization header.
-The Authorization Code Flow mechanism redirects the user to an OIDC provider to authenticate the identity of the user.
+The Quarkus `quarkus-oidc` extension provides a reactive, interoperable, multitenant-enabled OIDC adapter that supports Bearer and Authorization Code Flow authentication mechanisms.
+The Bearer authentication mechanism extracts the token from the HTTP Authorization header.
+The Authorization Code Flow mechanism redirects the user to an OIDC provider to authenticate the user's identity.
 After the user is redirected back to Quarkus, the mechanism completes the authentication process by exchanging the provided code that was granted for the ID, access, and refresh tokens.
 
-You can verify ID and access JWT tokens by using the refreshable JSON Web Key (JWK) set or you can introspect them remotely.
-However, opaque (binary) tokens can only be introspected remotely.
+You can verify ID and access JSON Web Key (JWK) tokens by using the refreshable JWK set or introspect them remotely.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-authentication-mechanisms-concept.adoc:182]:
```suggestion
You can verify ID and access JSON Web Token (JWT) tokens by using the refreshable JSON Web Key (JWK) set or introspect them remotely.
```

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-authentication-mechanisms-concept.adoc:182]: PR line changes the meaning

File: docs/src/main/asciidoc/security-authentication-mechanisms-concept.adoc [hunk 2]
```diff
@@ -150,64 +147,67 @@ X509Certificate certificate = credential.getCertificate();
...
+However, opaque, also known as binary tokens, can only be introspected remotely.
 
 [NOTE]
 ====
-Using the Quarkus OIDC extension, both Bearer Token and Authorization Code Flow mechanisms use <<smallrye-jwt-authentication, SmallRye JWT>> to represent JWT tokens as MicroProfile JWT `org.eclipse.microprofile.jwt.JsonWebToken`.
+Using the Quarkus OIDC extension, both Bearer Token and Authorization Code Flow mechanisms use xref:smallrye-jwt-authentication[SmallRye JWT authentication] to represent JWT tokens as MicroProfile JWT `org.eclipse.microprofile.jwt.JsonWebToken`.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-authentication-mechanisms-concept.adoc:187]:
```suggestion
Using the Quarkus OIDC extension, both Bearer and Authorization Code Flow authentication mechanisms use xref:smallrye-jwt-authentication[SmallRye JWT authentication] to represent JWT tokens as MicroProfile JWT `org.eclipse.microprofile.jwt.JsonWebToken`.
```

File: docs/src/main/asciidoc/security-authentication-mechanisms-concept.adoc [hunk 3]
```diff
@@ -218,36 +218,40 @@ The `quarkus-oidc-client` extension provides `OidcClient` for acquiring and refr
...
+This filter can be registered with MicroProfile REST client implementations injected into the current Quarkus endpoint, which must be protected by using the Quarkus OIDC adapter.
+This filter can propagate the access token to the downstream services.
 
 For more information, see the xref:security-openid-connect-client.adoc[OpenID Connect client and token propagation quickstart] and xref:security-openid-connect-client-reference.adoc[OpenID Connect (OIDC) and OAuth2 client and filters reference] guides.
 
 [[smallrye-jwt-authentication]]
 === SmallRye JWT authentication
 
-The `quarkus-smallrye-jwt` extension provides a MicroProfile JSON Web Token (JWT) 1.2.1 implementation and multiple options to verify signed and encrypted `JWT` tokens and represents them as `org.eclipse.microprofile.jwt.JsonWebToken`.
+The `quarkus-smallrye-jwt` extension provides a MicroProfile JSON Web Token (JWT) 1.2.1 implementation and multiple options to verify signed and encrypted `JWT` tokens.
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-authentication-mechanisms-concept.adoc:241]:
```suggestion
The `quarkus-smallrye-jwt` extension provides a MicroProfile JSON Web Token (JWT) 2.1 implementation and multiple options to verify signed and encrypted `JWT` tokens.
```

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-authentication-mechanisms-concept.adoc:241]: Sorry this is not this PR problem :-), but lets fix it now if you don't mind

**Comment by sberyozkin** [comment]
Michelle, @michelle-purcell, very nice updates, thanks, I suggested only a few minor changes

**Comment by michelle-purcell** [comment]
@sberyozkin - Thanks for your review and comments. I have fixed all of them :+1:

**Comment by sberyozkin** [comment]
Thanks @michelle-purcell, LGTM, one question I'd like to ask though first, do we really want to optimize from `Bearer token authentication` to `Bearer authentication` given that if we do it then I think we'll need to remove `-token-` from the corresponding file names and do more redirects to the website. I don't mind as long as users will find the new files without experiencing `404`, but,  `Bearer token authentication` is not that bad, tiny bit redundant but `Bearer token` is a well known concept.

What do you think ? I'm fine to go with `Bearer authentication` if you prefer, but I thought I'd double check

**Comment by michelle-purcell** [comment]
@sberyozkin - Apologies if I'm stomping on the same ground again... 

After researching other "official" sources, the official scheme name seems to be "Bearer authentication", however, there are lots of docs out there that refer to the same as "Bearer Token authentication" or "Bearer token authentication". 

 I suggest we use whatever the confirmed official term is for the "Bearer" authentication standard and be consistent throughout the docs, hence, I opted for:
`Bearer authentication` when referring to the name of this authentication mechanism or scheme.
`bearer token` when talking generically about bearer-type tokens.
 A "Bearer token" is also the correct form too, when it makes sense to compare specific token types or make a statement about a token that is only specific to Bearer. 
WDYT?

Regarding the filenames and IDs with `...bearer-token...` in them, because `token` is a good keyword here, I think renaming the URL and filename is not necessary. Keeping token as a keyword in the filename and ID  would be helpful for SEO.

I will do another pass at the other content to use the consistent terms above or whatever we agree on.

**Comment by sberyozkin** [comment]
Hi @michelle-purcell 

Thanks.

As far as the `bearer token` vs `Bearer token` is concerned, when we say something like `Multiple tenants that can support bearer token or authorization code flow mechanisms` - then the lower case does not quite work for me here, it is like we are talking about some individual tokens or authorization codes (it is not what you modified - I just see it in the text).

But, something like: `quarkus-oidc requires an OpenID Connect provider such as Keycloak, which can verify the bearer tokens` is fine - this your PR's line - here we are talking about individual tokens.

I'd not be worried about some general guidelines in this context, or how someone else refers to them somewhere else, `Bearer token`, `Authorization code flow` are well known concepts.

I propose to use this guideline: if we talk about the individual cases of the token verification, we should use `bearer token`, but a general concept is `Bearer token`.

As far as the file names are concerned: if we think `token` is a good keyword, why move from `Bearer token authentication` to `Bearer authentication` in the actual text ? The former is certainly correct and more descriptive ?

Thanks

**Comment by michelle-purcell** [comment]
@sberyozkin - Thanks for more context and clarification around this.  We are close :pray: I think...

So if we follow best practice to consistently use the capitalization and terminology of the 'thing' (proper noun) as defined by its creator/author/inventor, then in this case, then we should use `Bearer authentication mechanism` or `Bearer authentication scheme` to name the mechanism/scheme consistently.

Today we use a mix and match of both in Quarkus docs so readers could get confused that we are talking about 2 different things. Plus they could get translated differently and they diverge into 2 different things. 

We need to decide and agree on which one to use.
If you think that referring to the Bearer authentication mechanism as 'Bearer token authentication mechanism' will resonate more with Quarkus developers/users and is a common reference to it throughout the industry then let's use that, but call it out in the most prominent topic on the subject. (I notice Red Hat and other popular security content sometimes use this too but the other way is just as popular). 

I think the main point is that we are consistent throughout the docs.

I updated this topic a super above. If you agree, I'll fix other docs that contain 'Bearer token' content in later PRs too.

As for the filename bit... 
'Bearer token' is perfectly correct when Bearer is used as an adjective. It's used a lot in the docs and therefore makes a good keyword for the filename. A filename plays a part also in SEO, so it's more than about matching a title, which is Quarkus contributing practice. But if we agree on above then the filename comment is resolved.

WDYT? Thanks so much for persevering with this one!






 if the 'thing' is the [RFC-6750](https://datatracker.ietf.org/doc/html/rfc6750) we should use '




**Bearer authentication** mechanism is the official name defined in the RFC.

 hence the suggestion to use that consistently, which we don't in Quarkus docs today. We nee. However, in many docs,

**Comment by sberyozkin** [comment]
Hi Michelle @michelle-purcell,

Looks like we are in the agreement, `Bearer token authentication`, while a tiny bit more verbose, does not change the meaning of the shorter `Bearer authentication` but makes it a bit more explicit about the underlying  credential type involved, and like you mentioned, having `token` included in the file name (but also across the text) should only improve the searchability. 

FYI, `Bearer authentication scheme` (specifically, `scheme`) would usually be a technical reference to the content of the `Authoriization` header, the first word in `Authorization` header value identifies a `scheme` (Basic, Bearer, etc)...

`Bearer token authentication` looks like a good compromise to me, thanks 

>Thanks so much for persevering with this one!

You are too kind :-), it should be thanks to you for the unlimited patience when dealing with my review comments :-)

**Review by sberyozkin — APPROVED** [review]

**Comment by michelle-purcell** [comment]
Thanks @sberyozkin :+1:

**Comment by rsvoboda** [comment]
Hi @michelle-purcell. I reviewed the PR and have no objections to the changes.

**Pull request merged**